### PR TITLE
Detect num_classes information for the H-label classification

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

This PR introduces,
* Automatically takes the `hlabel_info` from the `Datamodule`  and changes the `num_multiclass_heads` and `num_multilabel_classes` information. So, after this PR, users don't need to manually set `--model.num_multiclass_heads` and `--model.num_multilabel_casses` which is quite harmful to UX.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — COMMENTED** [review]
One concern is that API users will eventually have to update `num_multiclass_heads` and `num_multilabel_classes` themselves.

Shouldn't we do this somewhere other than the CLI?
For example, I think this should be updated at some point when `lit_module.label_info = datamodule.label_info`.
If the metric is the problem, I think it would be better to create a new one when label_info is updated.

**Comment by harimkang** [comment]
> One concern is that API users will eventually have to update `num_multiclass_heads` and `num_multilabel_classes` themselves.
> 
> Shouldn't we do this somewhere other than the CLI? For example, I think this should be updated at some point when `lit_module.label_info = datamodule.label_info`. If the metric is the problem, I think it would be better to create a new one when label_info is updated.

**Comment by sungmanc** [comment]
> One concern is that API users will eventually have to update `num_multiclass_heads` and `num_multilabel_classes` themselves.
> 
> Shouldn't we do this somewhere other than the CLI? For example, I think this should be updated at some point when `lit_module.label_info = datamodule.label_info`. If the metric is the problem, I think it would be better to create a new one when label_info is updated.

I also think the API side that an user delievers the class instant via API (i.e. Engine.train(model=EfficientNetb0HLabelClassification, metric=MixedHLabelAccuracy)). It's hard to consider when we consider the delivery of class instant since the metric needs that information (num_multiclass_heads) at the init phase. I want to do that at the later phase.

File: src/otx/cli/cli.py [hunk 1]
```diff
@@ -422,6 +422,14 @@ def instantiate_model(self, model_config: Namespace) -> tuple:
                 warn(warning_msg, stacklevel=0)
                 model_config.init_args.num_classes = num_classes
 
+            # Hlabel classification
+            from otx.core.data.dataset.classification import HLabelInfo
```
Comment (kprokofi) [inline src/otx/cli/cli.py:426]: Can we move the import higher so as not to mix with the code? Either in this function or at the beginning of the file.

Comment (harimkang) [reply src/otx/cli/cli.py:426]: Unfortunately, the CLI doesn't assume torch or lightning is installed by default, so this is a good place to be.

**Review by jaegukhyun — APPROVED** [review]

**Comment by sungmanc** [comment]
@harimkang , @kprokofi , Thanks for the comments, do I need to fix or add something in this PR ?

**Review by harimkang — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

**Pull request merged**